checker/src/call_visitor.rs

```diff
@@ -1190,49 +1188,72 @@ impl<'call, 'block, 'analysis, 'compilation, 'tcx, E>
     #[logfn_inputs(TRACE)]
     fn handle_has_tag(&mut self, checking_presence: bool) {
         precondition!(self.actual_args.len() == 1);
-        let result: Rc<AbstractValue>;
-        if let Some(tag) = self.extract_tag_kind_and_mask() {
+
+        let result: Option<Rc<AbstractValue>>;
+        if let Some(tag) = self.extract_tag_kind_and_propagation_set() {
             let source_path = Path::new_deref(self.actual_args[0].0.clone())
```

Comment:
The deref and the refine seems unnecessary at best and possibly wrong.

Reply:
The argument of `add_tag!` and `has_tag!` is a reference. What is a correct way to dereference a path?

Reply:
OK, I guess it is complicated, see below.